daemon/config/config.go

```diff
@@ -71,6 +71,12 @@ type commonBridgeConfig struct {
 	FixedCIDR string `json:"fixed-cidr,omitempty"`
 }
 
+// NetworkConfig stores the daemon-wide networking configurations
+type NetworkConfig struct {
+	// Default address pools for docker networks
+	DefaultAddressPools *opts.PoolsOpt `json:"default-node-local-address-pools,omitempty"`
```

Comment:
Seems like this can be a value instead of a pointer, then we can skip the nil.

Reply:
As we discussed , I have added proper null check all the places I use it.

Reply:
I thought we discussed forgoing the pointer because the zero value is all we need to check?

Reply:
I thought we discussed on checking nil all the places and I mentioned we check all the places properly. Anyways I can change it.